src/cron.rs

```diff
@@ -131,54 +131,61 @@ pub fn update_nodes(
     }
 
     // Compute which nodes changed their state, also update node names in DB
+    let online_nodes = cur_nodes_map.values().filter(|data| data.online).count();
     let changed : Vec<(String, NodeData)> = db.transaction::<_, anyhow::Error, _>(|| {
         let mut changed = Vec::new();
 
-        // Go over every node in the database
-        let db_nodes = nodes::table.load::<models::NodeQuery>(db)?;
-        for db_node in db_nodes.into_iter() {
-            let (id, db_data) = model_to_node_data(db_node);
-            if let Some(cur_data) = cur_nodes_map.remove(&id) {
-                // We already know this node.
-                // Did it change?
-                if cur_data != db_data {
-                    // Update in database
-                    diesel::update(nodes::table.find(id.as_str()))
-                        .set((
-                            nodes::name.eq(cur_data.name.as_str()),
-                            nodes::online.eq(cur_data.online)
-                        ))
+        if online_nodes < config.ui.min_online_nodes_on_map {
```

Comment:
Why do you do this in the DB transaction? You're not even accessing the DB.

Better do this before `let changed`. And `return` in here to avoid rightwards drift.

Reply:
> And `return` in here to avoid rightwards drift.

what is "rightwards drift"?

Reply:
You had to indent all that code one more level as it is inside the `else`. That's rightwards drift: when the left half of your screen is empty because everything is indented so much.

Reply:
Ah okay, so without else block..... I will change that later then too

Reply:
Well, the shift makes review much harder as the diff is a lot bigger than the real change.